File: Sorting Plateau/Counting Sort_Merge Sort_Quickselect/Q1. Kth Largest Element in an Array/Q1_KthLargestElementinanArray.py
import heapq  # Import heapq module for using heap operations

class Solution(object):
    def findKthLargest(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        # Create a min-heap with the first k elements
        min_heap = nums[:k]
        heapq.heapify(min_heap)  # Heapify to maintain the heap property

        # Iterate through the rest of the numbers
        for num in nums[k:]:
            if num > min_heap[0]:        # If current num is bigger than heap root
                heapq.heappop(min_heap)  # Remove the smallest in heap
                heapq.heappush(min_heap, num)  # Push the current num to maintain size k

        return min_heap[0]  # The root of the heap is the kth largest element

File: Sorting Plateau/Counting Sort_Merge Sort_Quickselect/Q1. Kth Largest Element in an Array/Q1_KthLargestElementinanArray.java
import java.util.PriorityQueue;

class Solution {
    // Solution class

    public int findKthLargest(int[] nums, int k) {
        // Method to find the k-th largest element

        PriorityQueue<Integer> queue = new PriorityQueue<>();
        // Min-heap to store k largest elements

        for (int no : nums) {
            // Loop through each number in the array

            queue.add(no);
            // Add current number to the heap

            if (queue.size() > k) {
                // If heap size exceeds k

                queue.remove();
                // Remove the smallest element
            }
        }

        return queue.poll();
        // Return the k-th largest element
    }
}
